cfgov/v1/atomic_elements/atoms.py

```diff
@@ -154,3 +155,33 @@ def clean(self, data):
             raise ValidationError("ImageBasicUrlAlt validation errors", params=error_dict)
         else:
             return data
+
+class AnchorLink(blocks.StructBlock):
+    link_id = blocks.CharBlock(required=False)
+    # full_url = blocks.CharBlock(required=False, default=slugurl)
+
+    def clean(self, data):
+        error_dict = {}
+
+        def format_id(string):
+            words = string.split('_');
+            suffix = ''
+            if string:
+                suffix = '_'
+            if 'anchor' in words:
+                return slugify(string)
+            else:
+                return get_unique_id('anchor_' + slugify(string) + suffix)
```

Comment:
suffix appends an underscore to the string if the user entered some text. for example:

1. user enters 'video' for the anchor field
2. `format_id` adds the 'anchor_' prefix to the beginning of the string 'video', followed by an underscore (from suffix) and appends a unique id 

versus
1. user does not enter an input string for anchor link
2. field generates anchor_34938908908 as an ID when no custom text is entered, otherwise field would have 2 underscores